algorithms/estonian/stemmer.tt

```diff
@@ -0,0 +1,479 @@
+    <DT><B><I>jääksin  &nbsp;  jääksid  &nbsp;  jääks  &nbsp;  jääksime  &nbsp;  jääksite</I></B>
+        <DD>replace by <B><I>jääma</I></B>
+    <DT><B><I>jäime  &nbsp;  jäite  &nbsp;  jäin  &nbsp;  jäid  &nbsp;  jäi</I></B>
+        <DD>replace by <B><I>jääma</I></B>
+    <DT><B><I>müün  &nbsp;  müüd  &nbsp;  müüb  &nbsp;  müüs  &nbsp;  müüme  &nbsp;  müüte  &nbsp;  müüvad</I></B>
+        <DD>replace by <B><I>müüsi</I></B>
+    <DT><B><I>müüksin  &nbsp;  müüksid  &nbsp;  müüks  &nbsp;  müüksime  &nbsp;  müüksite</I></B>
+        <DD>replace by <B><I>müüsi</I></B>
+    <DT><B><I>müüakse  &nbsp;  müüdi  &nbsp;  müüa  &nbsp;  müüma  &nbsp;  müümata</I></B>
+        <DD>replace by <B><I>müüsi</I></B>
+    <DT><B><I>loeb  &nbsp;  loen  &nbsp;  loed  &nbsp;  loeme  &nbsp;  loete  &nbsp;  loevad</I></B>
+        <DD>replace by <B><I>luge</I></B>
+    <DT><B><I>loeks  &nbsp;  loeksin  &nbsp;  loeksid  &nbsp;  loeksime  &nbsp;  loeksite</I></B>
+        <DD>replace by <B><I>luge</I></B>
+    <DT><B><I>põen  &nbsp;  põeb  &nbsp;  põed  &nbsp;  põeme  &nbsp;  põete &nbsp;  põevad</I></B>
+        <DD>replace by <B><I>põde</I></B>
+    <DT><B><I>põeksin  &nbsp;  põeks  &nbsp;  põeksid  &nbsp;  põeksime  &nbsp;  põeksite</I></B>
+        <DD>replace by <B><I>põde</I></B>
+    <DT><B><I>laon  &nbsp;  laob  &nbsp;  laod  &nbsp;  laome  &nbsp;  laote  &nbsp;  laovad</I></B>
+        <DD>replace by <B><I>ladu</I></B>
+    <DT><B><I>laoksin  &nbsp;  laoks  &nbsp;  laoksid  &nbsp;  laoksime  &nbsp;  laoksite</I></B>
+        <DD>replace by <B><I>ladu</I></B>
+    <DT><B><I>teeksin  &nbsp;  teeks  &nbsp;  teeksid  &nbsp;  teeksime  &nbsp;  teeksite</I></B>
+        <DD>replace by <B><I>tegi</I></B>
+    <DT><B><I>teen  &nbsp;  teeb  &nbsp;  teed  &nbsp;  teeme  &nbsp;  teete  &nbsp;  teevad</I></B>
+        <DD>replace by <B><I>tegi</I></B>
+    <DT><B><I>tehakse  &nbsp;  tehti  &nbsp;  tegema  &nbsp;  tegemata  &nbsp;  teha</I></B>
+        <DD>replace by <B><I>tegi</I></B>
+    <DT><B><I>näen  &nbsp;  näeb  &nbsp;  näed  &nbsp;  näeme  &nbsp;  näete  &nbsp;  näevad</I></B>
+        <DD>replace by <B><I>nägi</I></B>
+    <DT><B><I>näeksin  &nbsp;  näeks  &nbsp;  näeksid  &nbsp;  näeksime  &nbsp;  näeksite</I></B>
+        <DD>replace by <B><I>nägi</I></B>
+    <DT><B><I>nähakse  &nbsp;  nähti  &nbsp;  näha  &nbsp;  nägema  &nbsp;  nägemata</I></B>
+        <DD>replace by <B><I>nägi</I></B>
+</DL>
+</DL>
+
+
+<p>
+Step 1: emphasis
+</p>
+
+<DL><DD>
+    Search for the longest among the following suffixes in <I>R1</I>, and perform the action indicated
+<BR><BR>
+<DL>
+        <DD>Test if there’s at least 4 letters before <I>R1</I> region. If so, continue this step
+    <DT><B><I>gi</I></B>
+        <DD>if preceded by <B><I>GI</I></B> and <B>not <I>LONGV</I></B>, delete
+    <DT><B><I>ki</I></B>
+        <DD>if preceded by <B>är</B>, replace it with <B>k</B>. If it wasn't replaced with <B>k</B> and is preceded by <B><I>KI</I></B>, delete
```

Comment:
The implementation doesn't seem to include the `if preceded by <B>är</B>, replace it with <B>k</B>` part.

Reply:
I found the repo where this was developed and this was present in an earlier version of the algorithm but removed before submission to us:

https://github.com/texta-tk/est-snowball/commit/a1cf0654ded4beead22499bc2a7649a77b83c038#diff-2b5cd3254584cdfe27ac5ebcbdbf1881eeb43a10612fe91d9aeeb19c4e4501f0L54-R55

Therefore it seems this should be updated to:

> if preceded by _KI_, delete